modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -763,6 +764,60 @@ static void PostOutputs(InferenceEngine::InferRequest   &request,
     }
 }
 
+class PostOutputsList {
+public:
+    PostOutputsList(size_t size,
+                    std::shared_ptr<IECallContext> ctx,
+                    std::vector<std::vector<int>>&& cached_dims);
+
+    void operator()(InferenceEngine::InferRequest &request) const;
+
+private:
+    struct Priv {
+        size_t              size;
+        std::atomic<size_t> finished{0u};
+        std::shared_ptr<IECallContext> ctx;
+        std::vector<std::vector<int>> cached_dims;
+    };
+    std::shared_ptr<Priv> m_priv;
```

Comment:
Do we really need this if class is only in implementation part?

Reply:
Yes, because it contains `atomic` which doesn't have `copy ctor`

Reply:
Another reason to implement it such way is: object should be shared between async requests